src/BenchmarkDotNet.Diagnostics.Windows/NativeMemoryDiagnoser.cs

```diff
@@ -0,0 +1,257 @@
+
+        public IEnumerable<IAnalyser> Analysers => Array.Empty<IAnalyser>();
+        public void DisplayResults(ILogger logger)
+        {
+            logger.WriteLineHeader(new string('-', 20));
+            foreach (var line in Logger.CapturedOutput)
+                logger.Write(line.Kind, line.Text);
+        }
+
+        public void Handle(HostSignal signal, DiagnoserActionParameters parameters)
+        {
+            etwProfiler.Handle(signal, parameters);
+        }
+
+        //Code is inspired by https://github.com/Microsoft/perfview/blob/master/src/PerfView/PerfViewData.cs#L5719-L5944
+        private (long totalMemory, long memoryLeak) ParseEtlFile(BenchmarkCase parameters)
+        {
+            (long totalMemory, long memoryLeak) result = (0, 0);
+
+            var traceFilePath = etwProfiler.BenchmarkToEtlFile[parameters];
+            var etlxFile = Path.ChangeExtension(traceFilePath, ".etlx");
+            var options = new TraceLogOptions();
+            TraceLog.CreateFromEventTraceLogFile(traceFilePath, etlxFile, options);
+
+            using (var eventLog = new TraceLog(etlxFile))
+            {
+                TraceEvents events = eventLog.Events;
+
+                var eventSource = events.GetSource();
+
+                var bdnEventsParser = new EngineEventLogParser(eventSource);
+
+                var start = false;
+                var isFirstActualStartEnd = false;
+
+                long totalOperation = 0;
+
+                bdnEventsParser.WorkloadActualStart += data =>
+                {
+                    if (!isFirstActualStartEnd)
+                    {
+                        start = true;
+                    }
+
+                    totalOperation = data.TotalOperations;
+                };
+                bdnEventsParser.WorkloadActualStop += data =>
+                {
+                    start = false;
+                    isFirstActualStartEnd = true;
+                };
+
+                var heapParser = new HeapTraceProviderTraceEventParser(eventSource);
+                // We index by heap address and then within the heap we remember the allocation stack
+                var heaps = new Dictionary<Address, Dictionary<Address, long>>();
+                Dictionary<Address, long> lastHeapAllocs = null;
+
+                Address lastHeapHandle = 0;
+
+                long cumMetric = 0;
+                long totalAllocation = 0;
+
+                heapParser.HeapTraceAlloc += delegate (HeapAllocTraceData data)
+                {
+                    if (!start)
+                    {
+                        return;
+                    }
+
+                    var allocs = lastHeapAllocs;
+                    if (data.HeapHandle != lastHeapHandle)
+                        allocs = GetHeap(data.HeapHandle, heaps, ref lastHeapAllocs, ref lastHeapHandle);
+
+                    allocs[data.AllocAddress] = data.AllocSize;
```

Comment:
should it be increasing, not overwriting the value?

```suggestion
                    allocs[data.AllocAddress] += data.AllocSize;
```

Reply:
I think that in this place we should overwrite the value. 

It is a new allocation on data.AllocAddress. You can not have multiple allocations on the same address. In `HeapTraceFree` event is `allocs.Remove(data.FreeAddress);`

There is original code from prefview:
https://github.com/microsoft/perfview/blob/234398a44fbfee5b828272b8201d19d10c7f3578/src/PerfView/PerfViewData.cs#L5760